File: asgn2_Inheritence_Part1/question.cpp
#include <string>
#include <iostream>
#include "question.h"

using namespace std;

/*------------------*/
Question::Question()
{
  questionText = "\0";
  answerText = "\0";
  correct = false;
}

/*-------------------------------*/
void Question::showQuestion()
{
  cout << questionText << "?" << endl;
}

/*---------------------------------------------*/
bool Question::checkAnswer(string givenAnswer)
{
  if (givenAnswer == answerText)
  {
    correct = true;
    return true;
  }
  
  correct = false;
  return false;
}

/*------------------------------------------*/
void Question::setQuestion(string questionText)
{
  this->questionText = questionText;
}

/*---------------------------------------*/
void Question::setAnswer(string answerText)
{
  this->answerText = answerText;
}

/*---------------------------------*/
void Question::showAnswer()
{
  cout << answerText << "." << endl;
}

/*------------------------------*/
void Question::markCorrect()
{
  if (correct)
  {
    cout << "Correct! Great job!" << endl;
  }
} 

File: Asgn6_JavaClasses/Question.java

/**
* Question.java
* An abstract class for any question type
* <p>
*
* @author  Alex Novitchkov
* @version 1.0
* @since   3/8/2020
*/

abstract public class Question
{
	protected String text;
	protected String answer;
	protected boolean correct;
	
	//------------------------------------------------------------------------
	/**
	 * Shows the question
	 */
	abstract public void showQuestion();
	
	//------------------------------------------------------------------------
	/**
	 * Shows the answer
	 */
	abstract public void showAnswer();
	
	//------------------------------------------------------------------------
	/**
	 * Checks parameter givenAnswer versus protected member answer
	 * @param givenAnswer, user inputed answer
	 * @return whether equal or not
	 */
	abstract public boolean checkAnswer(String givenAnswer);
	
	//------------------------------------------------------------------------
	/**
	 * Marks whether question was answered correctly
	 */
	abstract public void markCorrect();

	//------------------------------------------------------------------------
	/**
	 * Marks whether question was answered incorrectly
	 */
	abstract public void markIncorrect();
	
	//------------------------------------------------------------------------
	/**
	 * Checks whether the question's correct member is true
	 * @return true or false
	 */
	abstract public boolean isCorrect();
}
